src/abstractdataframe/selection.jl

```diff
@@ -125,11 +125,20 @@ end
 
 function select_transform!(nc::Pair{Int, Pair{ColRename, Symbol}},
                            df::AbstractDataFrame, newdf::DataFrame,
-                           transformed_cols::Dict{Symbol, Any}, copycols::Bool)
+                           transformed_cols::Dict{Symbol, Any}, copycols::Bool,
+                           allow_resizing_newdf::Base.RefValue{Bool})
     col_idx, (_, newname) = nc
     # it is allowed to request column tranformation only once
     @assert !hasproperty(newdf, newname)
+    # disallow shortening to 0 rows
+    if allow_resizing_newdf[] && nrow(newdf) == 1 && nrow(df) > 1
+        newdfcols = _columns(newdf)
+        for (i, col) in enumerate(newdfcols)
+            newdfcols[i] = fill(first(col), nrow(df))
```

Comment:
This is needed for `CategoricalArray` I think:
```suggestion
            newdfcols[i] = fill!(similar(col, nrow(df)), first(col))
```

Reply:
Indeed - it is always tricky.